sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -285,7 +285,7 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
         // compatible format, which means the data source is file-based and must have a `path`.
         require(table.storage.locationUri.isDefined,
           "External file-based data source table must have a `path` entry in storage properties.")
-        Some(new Path(table.location).toUri.toString)
```

Comment:
there's actually some discussion [about defining this properly](https://issues.apache.org/jira/browse/HADOOP-14217), related to the problem of "colons in object store paths". That's not going to help directly, but a sign of something which has historically been underspecified